server/protocols/jmap/src/main/java/org/apache/james/jmap/model/AttachmentAccessToken.java

```diff
@@ -0,0 +1,128 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public class AttachmentAccessToken implements SignedExpiringToken {
+
+    public static final String SEPARATOR = "_";
+
+    public static Builder builder() {
+        return new Builder();
+    }
+    
+    public static class Builder {
+        private String blobId;
+        private ZonedDateTime expirationDate;
+        private String signature;
+
+        private Builder() {}
+        
+        public Builder blobId(String blobId) {
+            this.blobId = blobId;
+            return this;
+        }
+
+        public Builder expirationDate(ZonedDateTime expirationDate) {
+            this.expirationDate = expirationDate;
+            return this;
+        }
+
+        public Builder signature(String signature) {
+            this.signature = signature;
+            return this;
+        }
+
+        public AttachmentAccessToken build() {
+            Preconditions.checkNotNull(blobId);
+            Preconditions.checkArgument(! blobId.isEmpty());
+            Preconditions.checkNotNull(expirationDate);
+            Preconditions.checkNotNull(signature);
+            return new AttachmentAccessToken(blobId, expirationDate, signature);
+        }
+    }
+    
+    private final String blobId;
+    private final ZonedDateTime expirationDate;
+    private final String signature;
+
+    @VisibleForTesting
+    AttachmentAccessToken(String blobId, ZonedDateTime expirationDate, String signature) {
+        this.blobId = blobId;
+        this.expirationDate = expirationDate;
+        this.signature = signature;
+    }
+
+    public String getBlobId() {
+        return blobId;
+    }
+
+    @Override
+    public ZonedDateTime getExpirationDate() {
+        return expirationDate;
+    }
+
+    @Override
+    public String getSignature() {
+        return signature;
+    }
+
+    public String serialize() {
+        return getContent()
+            + SEPARATOR
+            + signature;
+    }
+    
+    @Override
+    public String getContent() {
+        return DateTimeFormatter.ISO_OFFSET_DATE_TIME.format(expirationDate);
```

Comment:
Because the blobId is already sent in the GET request, so it is redundant to have it in the token.
After a little discussion, we decided to add the username as it is not present in the token and could be useful as no other authorization will be available.